woodchipper/context.py

```diff
@@ -0,0 +1,53 @@
+import contextvars
+from collections.abc import MutableMapping
+from decimal import Decimal
+from typing import Mapping, Optional, Union
+
+LoggableValue = Optional[Union[str, int, bool, Decimal, float]]
+LoggingContext = Mapping[str, LoggableValue]
+
+
+class LoggingContextVar(MutableMapping):
+    _var: contextvars.ContextVar[LoggingContext]
+
+    def __init__(self, name):
+        self._var = contextvars.ContextVar(name, default={})
+
+    def __getitem__(self, key: str) -> str:
+        return self._var.get({}).get(key)
+
+    def __setitem__(self, key: str, value: LoggableValue) -> contextvars.Token:
+        updated = self._var.get({}).copy()
+        updated[key] = value
+        return self._var.set(updated)
+
+    def __eq__(self, val) -> bool:
+        return val == self._var.get({})
+
+    def __delitem__(self, key: str) -> contextvars.Token:
+        updated = self._var.get({}).copy()
```

Comment:
[Using defaults in each of these cases will override the defaults that the context variables are initiated with](https://docs.python.org/3.8/library/contextvars.html?highlight=contextvar#contextvars.ContextVar.get).

Is that okay? Will we be initiating context outside of this lib -- will we want to?